### DISCUSSION TITLE
How can I prevent db connexion error

### DISCUSSION THREAD

**Discussion by MatteoGauthier** [prisma/prisma#11539] [Q&A] [answered]
Hey I build a project with Prisma and Postgres on Supabase, and the database sleeps after an amount of time. Is there a possibility to check the database connection with prisma, in order to display a message on the UI to notify backend error ?

Does someone know how to do this ? btw thanks for this amazing tool

**Comment by austincrim** [comment, marked as answer]
Hey @MatteoGauthier 👋 ,

[Per the Supabase docs](https://supabase.com/docs/going-into-prod#availability):
> If your application is on the free tier and is not expected to be queried at least once every 7 days, then it may be paused by Supabase to save on server resources.
> - you can restore paused projects from the Supabase dashboard
> - upgrade to Pro to guarantee that your project will not be paused for inactivit

It is not possible right now to check if the Prisma Client is connected to a database. You could try explicitly calling the [`$connect`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#connect-1) method and handling errors from there, but I haven't tried that approach yet.

**Reply by MatteoGauthier** [reply]
Okay thanks for your answer, the `$connect` function will do the thing. 👌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
